models/GroupsService.php

```diff
@@ -138,7 +145,54 @@ public function addUser($userUri, core_kernel_classes_Resource $group)
      */
     public function removeUser($userUri, core_kernel_classes_Resource $group)
     {
-        $user = new \core_kernel_classes_Resource($userUri);
-        return $user->removePropertyValue(new core_kernel_classes_Property(self::PROPERTY_MEMBERS_URI), $group);
+        return $this->getModel()->getResource($userUri)->removePropertyValue(
+            $this->getModel()->getProperty(self::PROPERTY_MEMBERS_URI),
+            $group
+        );
+    }
+
+    /**
+     * Duplicates a Group, copying associations for former  Test Takers and
+     * Deliveries to the new group.
+     *
+     * Test takers assigned to the group are not copied by the parent class
+     * method (but deliveries assigned to the group are), so we need to assign
+     * them here to the new group.
+     *
+     * @param core_kernel_classes_Resource $instance Group being cloned
+     * @param ?core_kernel_classes_Class $class Class to create the duplicate in
+     *
+     * @throws common_Exception
+     * @throws common_exception_Error
+     * @throws FileExistsException
+     *
+     * @return core_kernel_classes_Resource
+     */
+    public function cloneInstance(
+        core_kernel_classes_Resource $instance,
+        core_kernel_classes_Class $class = null
+    ) {
+        $newGroup = parent::cloneInstance($instance, $class);
+
+        foreach ($this->getUsers($instance->getUri()) as $user) {
+            $this->addUser($user->getUri(), $newGroup);
+        }
+
+        return $newGroup;
+    }
+
+    private function getTestTakerRootClass(): core_kernel_classes_Class
+    {
+        return $this->getTestTakerService()->getRootClass();
+    }
+
+    private function getTestTakerService(): TestTakerService
+    {
```

Comment:
hmmmmm i feel curious about this, maybe we should huddle, @hectoras

Reply:
This method has been removed :ok_hand: